src/openforms/submissions/tests/test_post_submission_event.py

```diff
@@ -1023,6 +1025,81 @@ def test_payment_complete_does_not_set_retry_flag(self):
 
             self.assertFalse(submission.needs_on_completion_retry)
 
+    @freeze_time("2024-02-16T18:15:00Z")
+    def test_successful_registration_updates_registration_counter_and_date(self):
+        submission = SubmissionFactory.from_components(
+            components_list=[
+                {
+                    "key": "email",
+                    "type": "email",
+                    "label": "Email",
+                    "confirmationRecipient": True,
+                }
+            ],
+            form__name="Pretty Form",
+            submitted_data={"email": "test@test.nl"},
+            completed_not_preregistered=True,
+            confirmation_email_sent=False,
+            form__registration_backend="email",
+            form__registration_backend_options={"to_emails": ["test@registration.nl"]},
+        )
+
+        with (
+            freeze_time("2024-02-16T21:15:00Z"),
+            patch(
+                "openforms.registrations.contrib.email.plugin.EmailRegistration.register_submission"
+            ) as mock_registration,
+        ):
+            on_post_submission_event(submission.id, PostSubmissionEvents.on_completion)
+
+        submission.refresh_from_db()
+
+        mock_registration.assert_called()
+        self.assertEqual(
+            submission.last_register_date,
+            datetime(2024, 2, 16, 21, 15).replace(tzinfo=timezone.utc),
+        )
+        self.assertEqual(submission.registration_attempts, 1)
+
+    @freeze_time("2024-02-16T18:15:00Z")
+    def test_preregister_success_and_registration_failure_updates_counter_and_date_only_once(
+        self,
+    ):
+        submission = SubmissionFactory.from_components(
+            components_list=[
+                {
+                    "key": "email",
+                    "type": "email",
+                    "label": "Email",
+                    "confirmationRecipient": True,
+                }
+            ],
+            form__name="Pretty Form",
+            submitted_data={"email": "test@test.nl"},
+            confirmation_email_sent=False,
+            registration_failed=True,
+            form__registration_backend="email",
+            form__registration_backend_options={},
+        )
+
+        with (
+            freeze_time("2024-02-16T21:15:00Z"),
+            patch(
+                "openforms.registrations.contrib.email.plugin.EmailRegistration.register_submission"
+            ) as mock_registration,
+        ):
+            on_post_submission_event(submission.id, PostSubmissionEvents.on_completion)
+
+        submission.refresh_from_db()
+
+        mock_registration.assert_not_called()
+        self.assertTrue(submission.pre_registration_completed)
```

Comment:
as discussed on Slack, I'd expect this to be:

```suggestion
        self.assertFalse(submission.pre_registration_completed)
```
since supposedly pre-registration fails in some manner

Reply:
Basically this test should succeed on the `pre_register` and fail on the `register`